src/pokemon_jump.c

```diff
@@ -3586,288 +3838,294 @@ static void sub_802DE1C(void)
+static void PrintPokeJumpPlayerName(int multiplayerId, u8 bgColor, u8 fgColor, u8 shadow)
 {
     u32 x;
-    u8 colors[3] = {clr1, clr2, clr3};
+    u8 colors[3] = {bgColor, fgColor, shadow};
 
-    FillWindowPixelBuffer(gUnknown_02022D00->unk1C[multiplayerId], 0);
-    x = 64 - GetStringWidth(1, sub_802C8E8(multiplayerId), -1);
+    FillWindowPixelBuffer(sPokemonJumpGfx->nameWindowIds[multiplayerId], 0);
+    x = 64 - GetStringWidth(1, GetPokeJumpPlayerName(multiplayerId), -1);
     x /= 2;
-    AddTextPrinterParameterized3(gUnknown_02022D00->unk1C[multiplayerId], 1, x, 1, colors, -1, sub_802C8E8(multiplayerId));
-    CopyWindowToVram(gUnknown_02022D00->unk1C[multiplayerId], 2);
+    AddTextPrinterParameterized3(sPokemonJumpGfx->nameWindowIds[multiplayerId], 1, x, 1, colors, -1, GetPokeJumpPlayerName(multiplayerId));
+    CopyWindowToVram(sPokemonJumpGfx->nameWindowIds[multiplayerId], 2);
 }
 
-static void sub_802DF70(bool32 arg0)
+static void PrintPokeJumpPlayerNames(bool32 highlightSelf)
 {
-    int i, var, playersCount = sub_802C8AC();
+    int i, multiplayerId, playersCount = GetNumPokeJumpPlayers();
 
-    if (!arg0)
+    if (!highlightSelf)
     {
         for (i = 0; i < playersCount; i++)
-            sub_802DED8(i, TEXT_COLOR_TRANSPARENT, TEXT_COLOR_DARK_GREY, TEXT_COLOR_LIGHT_GREY);
+            PrintPokeJumpPlayerName(i, TEXT_COLOR_TRANSPARENT, TEXT_COLOR_DARK_GREY, TEXT_COLOR_LIGHT_GREY);
     }
     else
     {
-        var = sub_802C8BC();
+        // Highlight own name
+        multiplayerId = GetPokeJumpMultiplayerId();
         for (i = 0; i < playersCount; i++)
         {
-            if (var != i)
-                sub_802DED8(i, TEXT_COLOR_TRANSPARENT, TEXT_COLOR_DARK_GREY, TEXT_COLOR_LIGHT_GREY);
+            if (multiplayerId != i)
+                PrintPokeJumpPlayerName(i, TEXT_COLOR_TRANSPARENT, TEXT_COLOR_DARK_GREY, TEXT_COLOR_LIGHT_GREY);
             else
-                sub_802DED8(i, TEXT_COLOR_TRANSPARENT, TEXT_COLOR_RED, TEXT_COLOR_LIGHT_RED);
+                PrintPokeJumpPlayerName(i, TEXT_COLOR_TRANSPARENT, TEXT_COLOR_RED, TEXT_COLOR_LIGHT_RED);
         }
     }
 }
 
-static void sub_802DFD4(void)
+static void DrawPlayerNameWindows(void)
 {
-    int i, playersCount = sub_802C8AC();
+    int i, playersCount = GetNumPokeJumpPlayers();
 
     for (i = 0; i < playersCount; i++)
-        PutWindowTilemap(gUnknown_02022D00->unk1C[i]);
-    CopyBgTilemapBufferToVram(0);
+        PutWindowTilemap(sPokemonJumpGfx->nameWindowIds[i]);
+    CopyBgTilemapBufferToVram(BG_INTERFACE);
 }
 
-static void sub_802E00C(u8 arg0)
+static void ShowBonus(u8 bonusId)
 {
-    gUnknown_02022D00->unk18 = 0;
-    ChangeBgX(1, (arg0 / 2) << 16, 0);
-    ChangeBgY(1, (((arg0 % 2) << 8) - 40) << 8, 0);
-    ShowBg(1);
-    CreateTask(sub_802E090, 4);
+    sPokemonJumpGfx->bonusTimer = 0;
+    ChangeBgX(BG_BONUSES, (bonusId / 2) << 16, 0);
+    ChangeBgY(BG_BONUSES, (((bonusId % 2) << 8) - 40) << 8, 0);
```

Comment:
Does it match with multiplication? Left-shifting negative integer is undefined behavior

Reply:
It does, updated (though `bonusId` is never negative here)